tests/lib/rules/no-undefined.js

```diff
@@ -42,6 +46,10 @@ ruleTester.run("no-undefined", rule, {
         { code: "function f(undefined) {}", errors },
         { code: "var undefined;", errors },
         { code: "try {} catch(undefined) {}", errors },
-        { code: "(function undefined(){}())", errors }
+        { code: "(function undefined(){}())", errors },
+        { code: "undefined = true", errors },
+        { code: "var undefined = true", errors },
+        { code: "({ bar: undefined })", errors },
+        { code: "({ bar: undefined } = foo)", errors }
```

Comment:
It might be useful to have tests for

```js
({ undefined }); // invalid

var { undefined } = foo; // invalid

({ undefined() {} }); // valid

({ undefined: function undefined() {} }); // valid

function undefined() {} // invalid, I think? It does create a variable

class Foo { undefined() {} } // valid

class Foo { [undefined]() {} } // invalid
```